code/game/sound.dm

```diff
@@ -23,120 +23,102 @@
-falloff_exponent - Rate of falloff for the audio. Higher means quicker drop to low volume. Should generally be over 1 to indicate a quick dive to 0 rather than a slow dive.
-frequency - playback speed of audio
-channel - The channel the sound is played at
-pressure_affected - Whether or not difference in pressure affects the sound (E.g. if you can hear in space)
-ignore_walls - Whether or not the sound can pass through walls.
-falloff_distance - Distance at which falloff begins. Sound is at peak volume (in regards to falloff) aslong as it is in this range.
-
-*/
-
+/**
+ * playsound is a proc used to play a 3D sound in a specific range. This uses SOUND_RANGE + extra_range to determine that.
+ *
+ * source - Origin of sound.
+ * soundin - Either a file, or a string that can be used to get an SFX.
+ * vol - The volume of the sound, excluding falloff and pressure affection.
+ * vary - bool that determines if the sound changes pitch every time it plays.
+ * extrarange - modifier for sound range. This gets added on top of SOUND_RANGE.
+ * falloff_exponent - Rate of falloff for the audio. Higher means quicker drop to low volume. Should generally be over 1 to indicate a quick dive to 0 rather than a slow dive.
+ * frequency - playback speed of audio.
+ * channel - The channel the sound is played at.
+ * pressure_affected - Whether or not difference in pressure affects the sound (E.g. if you can hear in space).
+ * ignore_walls - Whether or not the sound can pass through walls.
+ * falloff_distance - Distance at which falloff begins. Sound is at peak volume (in regards to falloff) aslong as it is in this range.
+ */
 /proc/playsound(atom/source, soundin, vol as num, vary, extrarange as num, falloff_exponent = SOUND_FALLOFF_EXPONENT, frequency = null, channel = 0, pressure_affected = TRUE, ignore_walls = TRUE, falloff_distance = SOUND_DEFAULT_FALLOFF_DISTANCE, use_reverb = TRUE)
 	if(isarea(source))
 		CRASH("playsound(): source is an area")
 
 	var/turf/turf_source = get_turf(source)
 
-	if (!turf_source)
+	if (!turf_source || !soundin || !vol)
 		return
 
 	//allocate a channel if necessary now so its the same for everyone
 	channel = channel || SSsounds.random_available_channel()
 
-	// Looping through the player list has the added bonus of working for mobs inside containers
 	var/sound/S = sound(get_sfx(soundin))
 	var/maxdistance = SOUND_RANGE + extrarange
 	var/source_z = turf_source.z
-	var/list/listeners = SSmobs.clients_by_zlevel[source_z].Copy()
+	var/list/listeners
+
+	. = list()//output everything that successfully heard the sound
 
 	var/turf/above_turf = SSmapping.get_turf_above(turf_source)
 	var/turf/below_turf = SSmapping.get_turf_below(turf_source)
 
-	if(!ignore_walls) //these sounds don't carry through walls
-		listeners = listeners & hearers(maxdistance,turf_source)
+	if(ignore_walls)
+		listeners = SSspatial_grid.orthogonal_range_search(turf_source, SPATIAL_GRID_CONTENTS_TYPE_CLIENTS, maxdistance)
```

Comment:
this was another cherry pick from a scrapped movement optimization pr and eh honestly i wasnt able to find any advantage to using the grid over iterating clients in the z level surprisingly enough, only thing i need here is returning real hearers